### ISSUE TITLE
Restrict access to cache endpoint

### ISSUE THREAD

**Issue by autumnfound** [EclipseFdn/marketplace-rest-api#13]
Currently, the cache endpoint is open to any user to make changes or send requests to. A mechanism will need to be added to restrict access either to users with an admin scope in OAuth or users that pass a secret access token to the server. If a secret token is implemented, the token should be configured via microprofile and be maintained within the secret properties file.

**Comment by mbarbero** [comment]
Another solution is to serve the cache endpoint on another port than the one being routed externally. It's a very common pattern for, e.g., metrics or liveness endpoints.

**Comment by autumnfound** [comment]
There doesn't seem to be any documentation around creating an application with multiple ports within Quarkus natively. I could look at adding custom code to listen on another port, but that is somewhat overkill.

**Comment by mbarbero** [comment]
You're right about Quarkus, this pattern does not look easy to implement (if at all). I was mentioning a common pattern in the microservices world, assuming it was easily doable with Quarkus. My bad!

**Comment by autumnfound** [comment]
I've seen the pattern before and it was a good point to bring up! Quarkus is still pretty young of a product offering, so some more fun or interesting patterns don't yet exist. We could look at proxy-passing or rerouting based on URL in nginx to obfuscate the ports instead as well, though that smacks of being a bit cludgy
